File: tools/email_examples/node/smtp_example.js
const nodemailer = require('nodemailer');

const host = process.env.SMTP_HOST || 'smtp.example.com';
const port = parseInt(process.env.SMTP_PORT || '587', 10);
const user = process.env.SMTP_USER;
const pass = process.env.SMTP_PASS;
const from = process.env.FROM_EMAIL || user;
const to = process.env.TO_EMAIL || 'jimyount1947@gmail.com';

async function send() {
  let transporter = nodemailer.createTransport({
    host,
    port,
    secure: port === 465,
    auth: user && pass ? { user, pass } : undefined,
  });

  const info = await transporter.sendMail({
    from,
    to,
    subject: 'Test email (nodemailer)',
    text: 'Hello from nodemailer example',
  });

  console.log('Message sent:', info.messageId);
}

send().catch((err) => { console.error(err); process.exit(1); });

File: tools/email_examples/python/smtp_example.py
import os
import smtplib
from email.message import EmailMessage

host = os.environ.get('SMTP_HOST', 'smtp.example.com')
port = int(os.environ.get('SMTP_PORT', 587))
user = os.environ.get('SMTP_USER')
password = os.environ.get('SMTP_PASS')
from_addr = os.environ.get('FROM_EMAIL', user)
to_addr = os.environ.get('TO_EMAIL', 'jimyount1947@gmail.com')

msg = EmailMessage()
msg['Subject'] = 'Test email (smtplib)'
msg['From'] = from_addr
msg['To'] = to_addr
msg.set_content('Hello from smtplib example')

with smtplib.SMTP(host, port) as s:
    if port == 465:
        s = smtplib.SMTP_SSL(host, port)
    else:
        s.ehlo()
        try:
            s.starttls()
        except Exception:
            pass

    if user and password:
        s.login(user, password)
    s.send_message(msg)
    print('Message sent')
